src/types/fetch.rs

```diff
@@ -119,4 +122,24 @@ impl Fetch {
             })
             .next()
     }
+
+    /// Extract the `INTERNALDATE` of a `FETCH` response
+    ///
+    /// See [section 2.3.3 of RFC 3501](https://tools.ietf.org/html/rfc3501#section-2.3.3) for
+    /// details.
+    pub fn internal_date(&self) -> Option<DateTime<FixedOffset>> {
+        self.fetch
+            .iter()
+            .filter_map(|av| match av {
+                AttributeValue::InternalDate(str) => Some(*str),
```

Comment:
I'm surprised that `str` is a valid variable identifier here since it's also a type. Maybe make it `s` or `date` instead so the syntax highlighter doesn't remain confused? Minor point though :)

Reply:
Well I remember that I just copy/pasted from your code, but I don't find the same in yours. I will fix it.